state/futures/actions.ts

```diff
@@ -1321,3 +1321,18 @@ const monitorAndAwaitTransaction = async (
 	await tx.wait();
 	dispatch(updateTransactionStatus(TransactionStatus.Confirmed));
 };
+
+export const getClosePositionOrderFee = createAsyncThunk<Wei, void, ThunkConfig>(
+	'futures/getClosePositionOrderFee',
+	async (_, { getState, extra: { sdk } }) => {
+		const state = getState();
+		const position = selectPosition(state);
+		const marketInfo = selectMarketInfo(state);
+
+		if (marketInfo) {
+			return sdk.futures.getOrderFee(marketInfo.market, position?.position?.size.neg() ?? zeroBN);
+		}
+
+		return zeroBN;
```

Comment:
probably want to throw an error if no market, can just pass a message to notifyError.

Reply:
Okay. I passed a message to notifyError in the latest changes, but was wondering if you think actually doing `throw new Error` is better.

Reply:
Yea I think we should throw as we are handling the error in the component in this case it seems through queryStatus. As the actual error details aren't passed to the reducer for handling the error state, usually I wrap the whole thing in a try catch, capture the error, pass it to notify and then continue to throw so the state still gets captured in the reducer.

Reply:
The thing is there is no actual error in this case though (it's just an undefined value), so wrapping this in try/catch won't work properly. Just to confirm, are you referring to catching any error that might be generated by the `getOrderFee` function or just if the `marketInfo` value is falsy?

Reply:
An example of what I was thinking is:

```
const state = getState();
const position = selectPosition(state);
const marketInfo = selectMarketInfo(state);
try {
  if (marketInfo) {
    return sdk.futures.getOrderFee(marketInfo.market, position?.position?.size.neg() ?? zeroBN);
  } else {
    throw new Error('No market found')
  }
} catch (err) {
  notifyError('Failed to get order fee', err)
  throw err
}
```

Reply:
Ah, I see. Got confused on the nesting part.